app/views/kaminari/_paginator.html.erb

```diff
@@ -0,0 +1,17 @@
+<%= paginator.render do %>
+  <nav>
+    <ul class="pagination">
+      <%= first_page_tag unless current_page.first? %>
+      <%= prev_page_tag unless current_page.first? %>
+      <% each_page do |page| %>
+        <% if page.left_outer? || page.right_outer? || page.inside_window? %>
```

Comment:
what is the function of this logic?

Reply:
This logic is implemented by the Kaminari gem, to perform the jumps of the pagination; thus to know on which page you are, and if there are pages on the sides to generate the views of these.